lib/reply.js

```diff
@@ -244,6 +248,47 @@ Reply.prototype.headers = function (headers) {
   return this
 }
 
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Trailer#directives
+// https://httpwg.org/specs/rfc7230.html#chunked.trailer.part
+const INVALID_TRAILERS = new Set([
+  'transfer-encoding',
+  'content-length',
+  'host',
+  'cache-control',
+  'max-forwards',
+  'te',
+  'authorization',
+  'set-cookie',
+  'content-encoding',
+  'content-type',
+  'content-range',
+  'trailer'
+])
+
+Reply.prototype.trailer = function (key, fn) {
+  const _key = key.toLowerCase()
+  if (INVALID_TRAILERS.has(_key)) {
+    throw new FST_ERR_BAD_TRAILER_NAME(_key)
+  }
+  if (typeof fn !== 'function') {
+    throw new FST_ERR_BAD_TRAILER_VALUE(_key, typeof fn)
+  }
+  if (this[kReplyTrailers] === null) this[kReplyTrailers] = {}
+  this[kReplyTrailers][_key] = fn
+  return this
+}
+
+Reply.prototype.hasTrailer = function (key) {
+  if (this[kReplyTrailers] === null) return false
+  return this[kReplyTrailers][key.toLowerCase()] !== undefined
+}
+
+Reply.prototype.removeTrailer = function (key) {
+  if (this[kReplyTrailers] === null) return this
+  this[kReplyTrailers][key.toLowerCase()] = undefined
```

Comment:
Even tho this is faster, Node.js does not like headers with undefined value, see: https://github.com/fastify/fastify/blob/802e3e8935dd8a84e08330faca83b7f7a108dad2/lib/reply.js#L208-L213

Reply:
They are not the same case, `trailer` is not a value. It is a function that return value.
Reassigning `undefined` is better in this case.

Reply:
You are right! My bad :)